### PR TITLE
acquisitions: account selector widget

### PR DESCRIPTION
**Pull request by zannkukai**
Creates a formly widget especially for `AcqAccount` resource. This
widget will generate a hierarchical dropdown list with a remaining
balance statement.

image

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Review by iGormilhit — COMMENTED** [review]
Commit title: s/acquisition: account selector widget/acquisitions: account selector widget/ (with an 's' to acquisitions).

File: projects/admin/src/app/acquisition/api/acq-account-api.service.ts [hunk 1]
```diff
@@ -98,4 +98,29 @@ export class AcqAccountApiService {
...
   }
 
+
+  /**
+   * Allow to sort accounts to render it correctly (corresponding to hierarchical tree structure)
+   * @param accounts - the accounts to sort.
+   * @return Account list sorted as a hierarchical tree.
+   */
+  orderAccountsAsTree(accounts: Array<AcqAccount>): Array<AcqAccount> {
+    /** Append an account and children accounts into the `accounts` list */
```
Comment (Garfield-fr) [inline projects/admin/src/app/acquisition/api/acq-account-api.service.ts:108]: cosmetic: `//` for one line

File: projects/admin/src/app/acquisition/components/editor/widget/select-account-editor-widget/select-account-editor-widget.component.html [hunk 2]
```diff
@@ -0,0 +1,51 @@
+<!--
+  RERO angular core
```
Comment (Garfield-fr) [inline projects/admin/src/app/acquisition/components/editor/widget/select-account-editor-widget/select-account-editor-widget.component.html:2]: `RERO ILS UI`

File: projects/admin/src/app/acquisition/components/editor/widget/select-account-editor-widget/select-account-editor-widget.component.html [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        <div class="pr-5">
+          {{ account.depth > 0 ? '&nbsp;&nbsp;&nbsp;&nbsp;'.repeat(account.depth) + ' ' : '' }}
+          {{ account.name }}
+        </div>
+        <div class="ml-auto pl-5 amount" [class]="{
+            'text-success': account.remaining_balance.self > 0,
+            'text-muted': account.remaining_balance.self === 0,
+            'text-warning': account.remaining_balance.self < 0
+        }">
+          {{ account.remaining_balance.self  | currency: organisation.default_currency }}
```
Comment (Garfield-fr) [inline projects/admin/src/app/acquisition/components/editor/widget/select-account-editor-widget/select-account-editor-widget.component.html:43]: Remove double space after `self`

File: projects/admin/src/app/acquisition/components/editor/widget/select-account-editor-widget/select-account-editor-widget.component.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
+/*
+ * RERO angular core
```
Comment (Garfield-fr) [inline projects/admin/src/app/acquisition/components/editor/widget/select-account-editor-widget/select-account-editor-widget.component.ts:2]: `RERO ILS UI`

**Review by Garfield-fr — APPROVED** [review]

File: projects/admin/src/app/acquisition/api/acq-account-api.service.ts (hunk 1 above)
Comment (zannkukai) [reply projects/admin/src/app/acquisition/api/acq-account-api.service.ts:108]: This is a doctring for inner function ;-)

**Pull request merged**